sdk/helper/keysutil/policy.go

```diff
@@ -1973,3 +1969,153 @@ func (p *Policy) EncryptWithFactory(ver int, context []byte, nonce []byte, value
 
 	return encoded, nil
 }
+
+func (p *Policy) UpdateKeyVersion(ctx context.Context, storage logical.Storage, key []byte, isPrivateKey bool, keyVersion int) error {
```

Comment:
The function renaming sounds good, will do it. About the nits:
- You are right, it wasn't really necessary. Moving the checks to the caller and using the actual key (source of truth) to determine the key is better and reduces the code.
- At the time of implementation I also thought about that possibility, however, I assumed that it would be better to give some kind of feedback to the user in cases like these as I believe an user wouldn't want to import on purpose the same key on a version twice.

Reply:
I'm fine leaving it returning an error if it is the same key for now.